spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubEmulatorConfiguration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.pubsub;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * 
+ * This Configuration auto detects a local pub/sub emulator environment and configures the
+ * client. To enable this auto-configuration the <code>PUBSUB_EMULATOR_HOST</code>
+ * property needs to be present.
+ *
+ * @author Andreas Berger
+ */
+@Configuration
+@ConditionalOnProperty("PUBSUB_EMULATOR_HOST")
+@AutoConfigureBefore(GcpContextAutoConfiguration.class)
```

Comment:
Shouldn't it be `@Before(GcpPubSubAutoConfiguration.class)`?

Reply:
I just used the same construct as you did in `org.springframework.cloud.gcp.autoconfigure.pubsub.GcpPubSubAutoConfiguration`

Reply:
Nevermind, you also need to inject the credentials, which is why it needs to be before the core auto-config and not Pub/Sub.